File: python/src/sikaru_api/types/tool_provider_ref_request.py

import typing

import pydantic
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel


class ToolProviderRefRequest(UniversalBaseModel):
    capability_prefix: str
    source_skill_refs: typing.Optional[typing.List[str]] = None
    tool_provider_id: str

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow

File: ruby/lib/Sikaru/types/tool_provider_ref_request.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class ToolProviderRefRequest < Internal::Types::Model
      field :capability_prefix, -> { String }, optional: false, nullable: false

      field :source_skill_refs, -> { Internal::Types::Array[String] }, optional: true, nullable: false

      field :tool_provider_id, -> { String }, optional: false, nullable: false
    end
  end
end
